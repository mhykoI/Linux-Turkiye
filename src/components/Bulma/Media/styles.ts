import { Styles } from 'react-jss';

const styles: Styles = {
  image: {
    borderRadius: '50%',
  },
};

export default styles;
