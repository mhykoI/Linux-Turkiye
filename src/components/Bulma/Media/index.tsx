import React from 'react';
import clsx from 'clsx';
import { createUseStyles } from 'react-jss';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface MediaProps {
  left?: {
    imgUrl?: string;
    size: 16 | 32 | 64 | 128 | 256;
  };
  right?: React.ReactNode;
}

const Media: React.FC<MediaProps> = ({ left, right, children }) => {
  const classes = useStyles();

  return (
    <article className="media">
      {left && (
        <figure className="media-left">
          <p className={clsx('image', `is-${left.size}x${left.size}`)}>
            <img
              className={classes.image}
              src={
                left.imgUrl ||
                'https://bulma.io/images/placeholders/128x128.png'
              }
            />
          </p>
        </figure>
      )}

      <div className="media-content">
        <div className="content">{children}</div>
      </div>

      {right && <div className="media-right">{right}</div>}
    </article>
  );
};

export default Media;
