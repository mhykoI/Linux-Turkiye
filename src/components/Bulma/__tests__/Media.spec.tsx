import React from 'react';
import renderer from 'react-test-renderer';
import Media from '../Media';

describe('<Media />', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Media
          left={{
            size: 64,
          }}
          right={<span>Hi!</span>}
        >
          This is a Media element!
        </Media>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
