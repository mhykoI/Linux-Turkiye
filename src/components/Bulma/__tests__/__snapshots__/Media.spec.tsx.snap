// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Media /> should render correctly 1`] = `
<article
  className="media"
>
  <figure
    className="media-left"
  >
    <p
      className="image is-64x64"
    >
      <img
        className="image-0-2-1"
        src="https://bulma.io/images/placeholders/128x128.png"
      />
    </p>
  </figure>
  <div
    className="media-content"
  >
    <div
      className="content"
    >
      This is a Media element!
    </div>
  </div>
  <div
    className="media-right"
  >
    <span>
      Hi!
    </span>
  </div>
</article>
`;
